File: moving_zeroes/movingZeroes.ts
export const movingZeroes = (arr: number[]) => {
  let idx = 0;
  let moved = 0;

  while (idx < arr.length - moved) {
    if (arr[idx] === 0) {
      arr.splice(idx, 1);
      arr.push(0);
      moved++;
    } else {
      idx++;
    }
  }
  return arr;
};

File: algorithms/moving_zeroes/moving_zeroes.py
def moving_zeroes(arr):
    idx = 0
    moved = 0

    while idx < len(arr) - moved:
        if arr[idx] == 0:
            arr.pop(idx)
            arr.append(0)
            moved += 1
        else:
            idx += 1

    return arr
